File: problemas/matematica/1630/1630.lua
function mdc(a, b)
  if b == 0 then
    return a
  end
  return mdc(b, a % b)
end

while true do
  linha = io.read()
  if linha == nil then
    break
  end

  numeros = {}
  for palavra in linha:gmatch("%S+") do
    numeros[#numeros + 1] = tonumber(palavra, 10)
  end
  x = numeros[1]
  y = numeros[2]

  print(math.floor((2 * (x + y)) / mdc(x, y)))
end

File: problemas/matematica/1630/1630.cpp
#include <iostream>

using namespace std;

int mdc(int x, int y){
    if(y == 0)  return x;
    else        return mdc(y, x % y);
}

int main(){
    int X, Y;

    while(cin >> X >> Y){
        if(X == Y)  cout << 4 << endl;
        else        cout << 2*(X + Y)/mdc(X, Y) << endl;
    }

    return 0;
}